src/core/thread/mesh_forwarder.cpp

```diff
@@ -911,8 +910,8 @@ ThreadError MeshForwarder::HandleFrameRequest(Mac::Frame &aFrame)
     // already checked and handled in `SendFragment()` method.
 
     if (((child = mNetif.GetMle().GetChild(macDest)) != NULL)
-        && ((child->mMode & Mle::ModeTlv::kModeRxOnWhenIdle) == 0)
-        && (child->GetQueuedMessageCount() > 1))
+        && !child->GetRxOnWhenIdle()
```

Comment:
nit: I know `RxOnWhenIdle` terminology follows the spec strictly, and originates from IEEE 802.15.4-2006 of old, but I feel like `child->IsSleepy()` would be more readable.